File: api_doc/hospital/api/models/ServiceModel.php
<?php
// api/models/ServiceModel.php

require_once __DIR__ . '/BaseModel.php';

class ServiceModel extends BaseModel {
    protected string $table = 'services';

    public function findAllActive(): array {
        return $this->query(
            "SELECT s.*, CONCAT(u.nom, ' ', u.prenom) AS responsable_nom
             FROM services s
             LEFT JOIN users u ON u.id = s.responsable_id
             WHERE s.is_active = 1
             ORDER BY s.nom ASC"
        );
    }

    public function findWithStats(int $id): ?array {
        $sql = "SELECT s.*,
                       CONCAT(u.nom, ' ', u.prenom) AS responsable_nom,
                       (SELECT COUNT(*) FROM employes e WHERE e.service_id = s.id AND e.statut = 'ACTIF') AS nb_employes,
                       (SELECT COUNT(*) FROM users us WHERE us.service_id = s.id AND us.deleted_at IS NULL) AS nb_utilisateurs
                FROM services s
                LEFT JOIN users u ON u.id = s.responsable_id
                WHERE s.id = ?";
        $stmt = $this->db->prepare($sql);
        $stmt->execute([$id]);
        return $stmt->fetch() ?: null;
    }

    public function findByType(string $type): array {
        $stmt = $this->db->prepare(
            "SELECT * FROM services WHERE type = ? AND is_active = 1 ORDER BY nom"
        );
        $stmt->execute([$type]);
        return $stmt->fetchAll();
    }

    public function getBudgetConsommation(int $serviceId, int $exerciceId): array {
        $sql = "SELECT b.montant_prevu,
                       pc.numero_compte, pc.libelle AS compte_libelle,
                       COALESCE(SUM(l.debit), 0) AS consomme,
                       b.montant_prevu - COALESCE(SUM(l.debit), 0) AS restant
                FROM budgets b
                JOIN plan_comptable pc ON pc.id = b.compte_id
                LEFT JOIN lignes_ecriture l ON l.compte_id = b.compte_id
                LEFT JOIN ecritures_comptables e ON e.id = l.ecriture_id
                    AND e.exercice_id = b.exercice_id
                    AND e.statut = 'VALIDE'
                WHERE b.service_id = ?
                  AND b.exercice_id = ?
                  AND b.statut = 'APPROUVE'
                GROUP BY b.id, pc.numero_compte, pc.libelle, b.montant_prevu";
        $stmt = $this->db->prepare($sql);
        $stmt->execute([$serviceId, $exerciceId]);
        return $stmt->fetchAll();
    }
}

File: hospital_nk/lib/data/models/service_model.dart
class ServiceModel {
  final int id;
  final String nom;
  final String code;
  final String? type;
  final Map<String, dynamic>? responsable;
  final double? budgetAnnuel;
  final int? nbEmployes;
  final String statut;
  final String? description;

  ServiceModel({
    required this.id,
    required this.nom,
    required this.code,
    this.type,
    this.responsable,
    this.budgetAnnuel,
    this.nbEmployes,
    required this.statut,
    this.description,
  });

  String? get responsableNom {
    if (responsable == null) return null;
    return '${responsable!['prenom'] ?? ''} ${responsable!['nom'] ?? ''}'.trim();
  }

  factory ServiceModel.fromJson(Map<String, dynamic> json) {
    return ServiceModel(
      id: json['id'] ?? 0,
      nom: json['nom'] ?? '',
      code: json['code'] ?? '',
      type: json['type'],
      responsable: json['responsable'],
      budgetAnnuel: json['budget_annuel']?.toDouble(),
      nbEmployes: json['nb_employes'],
      statut: json['statut'] ?? 'ACTIF',
      description: json['description'],
    );
  }

  Map<String, dynamic> toJson() => {
        'nom': nom,
        'code': code,
        'type': type,
        'responsable_id': responsable?['id'],
        'budget_annuel': budgetAnnuel,
        'description': description,
      };
}
